src/c++11/impl.cc

```diff
@@ -230,7 +231,11 @@ void SpanImpl::FinishSpan(SpanFinishOptions opts) {
   auto tags = span.mutable_tags();
 
   if (ref_.valid()) {
-    *tags->Add() = util::make_kv(ParentSpanGUIDKey, ref_.referenced().span_id());
+    // Note: Converting span_id to a hex string here is necessary
+    // because the Proto->Thrift conversion does not know to format
+    // int-valued tags as hex-valued, yet the string value for
+    // parent_span_guid is interpreted as a hex value downstream.
+    *tags->Add() = util::make_kv(ParentSpanGUIDKey, uint64ToHex(ref_.referenced().span_id()));
```

Comment:
1: I see, but checked the collector-side code and it could cause trouble
2: The uint64/string conversions will still be used for EnvoyCarrier, at present.